### PR TITLE
 Add install package to nuget GlobalPackagesFolder

### PR DESCRIPTION
**Pull request by wli3**
Allow separate asset.json folder in PackageInstance and make read actual package location from assetjson.

https://github.com/dotnet/cli/issues/9921

### CODE REVIEW

**Comment by wli3** [comment]
@dotnet-bot test this please

**Comment by wli3** [comment]
ping @dotnet/cli-contrib

File: src/dotnet/ToolPackage/IToolPackageInstaller.cs [hunk 1]
```diff
@@ -14,5 +14,10 @@ internal interface IToolPackageInstaller
             VersionRange versionRange = null,
             string targetFramework = null,
             string verbosity = null);
+
+        IToolPackage InstallPackageToExternalManagedLocation(PackageLocation packageLocation, PackageId packageId,
```
Comment (peterhuene) [inline src/dotnet/ToolPackage/IToolPackageInstaller.cs]: Nit: could we place the two parameters on their own lines like the remaining parameters (can change the `InstallPackage` above).  I don't like having multiple on one line and then only single parameters on remaining lines.

Nit: Could this simply be named `InstallExternalPackage`?  Just a suggestion, though.

File: src/dotnet/ToolPackage/ToolPackageInstaller.cs [hunk 2]
```diff
@@ -112,10 +116,48 @@ internal class ToolPackageInstaller : IToolPackageInstaller
                 });
         }
 
+        public IToolPackage InstallPackageToExternalManagedLocation(PackageLocation packageLocation,
+            PackageId packageId,
+            VersionRange versionRange = null,
+            string targetFramework = null,
+            string verbosity = null)
+        {
+            var stageDirectory = _store.GetRandomStagingDirectory();
```
Comment (peterhuene) [inline src/dotnet/ToolPackage/ToolPackageInstaller.cs]: So in `InstallPackage`, we do ensure the staging directory gets deleted when a rollback occurs.  Is that something we want for this method?

**Comment by wli3** [comment]
@dotnet-bot test opensuse.42.3 x64 Debug Build

**Comment by wli3** [comment]
@dotnet-bot test opensuse.42.3 x64 Debug Build

**Comment by wli3** [comment]
@dotnet-bot test opensuse.42.3 x64 Debug Build

**Comment by wli3** [comment]
@dotnet-bot test RHEL6 x64 Debug Build 
@dotnet-bot test Ubuntu16.04 x64 Debug Build
@dotnet-bot test ubuntu.18.04 x64 Debug Build

**Comment by wli3** [comment]
CI is weird today. Some machine just paused

File: src/dotnet/ToolPackage/ToolPackageInstaller.cs (hunk 2 above)
Comment (wli3) [reply src/dotnet/ToolPackage/ToolPackageInstaller.cs]: Good catch. I realize this is not used, so I removed them.

**Pull request merged**